components/expression/formula.jsx

```diff
@@ -0,0 +1,143 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Expression Condition design pattern](https://lightningdesignsystem.com/components/visual-picker/) in React.
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import assign from 'lodash.assign';
+
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+
+import { EXPRESSION_FORMULA } from '../../utilities/constants';
+
+import Button from '../button';
+
+const propTypes = {
+	assistiveText: PropTypes.shape({
+		help: PropTypes.string,
+	}),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * CSS classes to be added to tag with `.slds-form-element`. Uses `classNames` [API](https://github.com/JedWatson/classnames).
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+
+	labels: PropTypes.shape({
+		label: PropTypes.string,
+		checkSyntax: PropTypes.string,
+	}),
+
+	resourceCombobox: PropTypes.node,
+	functionCombobox: PropTypes.node,
+	operatorInput: PropTypes.node,
+
+	onClickHelp: PropTypes.func,
+	onClickCheckSyntax: PropTypes.func,
```

Comment:
Add two props:
- `textAreaValue`, of `PropTypes.node`. Notice the `node` type here; that's because a rich text editor might have special formatting beyond what a string can provide